src/OrleansRuntime/ReminderService/LocalReminderService.cs

```diff
@@ -310,32 +323,37 @@ private void PromoteToStarted()
             }
         }
 
-        private async Task ReadTableAndStartTimers(IRingRange range)
+        private async Task ReadTableAndStartTimers(IRingRange range, int cachedLocalRangeSequence)
         {
             if (logger.IsVerbose) logger.Verbose("Reading rows from {0}", range.ToString());
             localTableSequence++;
-            long cachedSequence = localTableSequence;
+            long cachedLocalTableSequence = localTableSequence;
```

Comment:
if we go with that solution, record the cachedLocalTableSequence next to where the range changes.